ebuild-writing/using-eclasses/text.xml

```diff
@@ -28,31 +28,32 @@ link="::general-concepts/portage-cache"/>).
 <p>
 After inheriting an eclass, its provided functions can be used as
 normal. Here's an example ebuild, <c>foomatic-0.1-r2.ebuild</c>, which
-uses four eclasses:
+uses three eclasses:
 </p>
 
 <codesample lang="ebuild">
 # Copyright 1999-2018 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=2
+EAPI=7
 
-inherit eutils bash-completion flag-o-matic autotools
+inherit autotools bash-completion-r1 flag-o-matic
 
 DESCRIPTION="Tool for foo"
-HOMEPAGE="http://foomatic.sf.net"
+HOMEPAGE="https://foomatic.sf.net"
 SRC_URI="mirror://sourceforge/${PN}/${P}.tar.gz"
 
 LICENSE="GPL-2"
 SLOT="0"
 KEYWORDS="alpha ~amd64 ~x86 ~x86-fbsd"
 
-RDEPEND=">=sys-libs/ncurses-5.2
-	>=sys-libs/readline-4.1"
+RDEPEND="sys-libs/ncurses:0=
+	>=sys-libs/readline:0="
 DEPEND="${RDEPEND}"
 
 src_prepare() {
-	epatch "${FILESDIR}/${P}-gentoo.patch"
+	default
+	eapply "${FILESDIR}/${P}-gentoo.patch"
```

Comment:
no, it is up to the maintainer

Reply:
But please change order. Manual patches should come before user patches.

Reply:
There should never be "default" and explicit "eapply" at the same time. Otherwise, the reader would have to check if there isn't a PATCHES variable.

So the example should either be:
- "default" along with PATCHES, or
- explicit "eapply" and "eapply_user".

Reply:
> But please change order. Manual patches should come before user patches which we will pull in via "default"..

Not convinced about that. If that involves Gentoo-specific patches, then applying user patches to vanilla sources may actually be better.

Reply:
Think about it again. If there are no conflicts between the Gentoo patch and the user patch, then the order doesn't matter. However, if there _are_ conflicts, then the user patch will cause the Gentoo patch to fail. In other words, by applying the user patch first, we would prevent the user from making changes on top of the Gentoo specific sources.

So, in general the order should be:
1. eapply
2. eapply_user
3. (if necessary) eautoreconf or similar

Reply:
Think about it again. If there are no conflicts between the Gentoo patch and the user patch, then the order doesn't matter. However, if there _are_ conflicts, then the user patch will cause the Gentoo patch to fail. In other words, by applying the user patch first, we would prevent the user from making changes on top of the Gentoo specific sources.

So, in general the order should be:
1. eapply
2. eapply_user
3. (if necessary) eautoreconf or similar

Reply:
@ulm, <snip>...

as was discussed at #g-qa after origanization shutdown (so I post it only now). I would want to leave `default` calls as they are now, as the main focus of the page is the eclasses usage, not the phases. Also we have the samples that explain what is `default` in the devmanual, but almost do not have examples with its usage.